index.js

```diff
@@ -55,6 +62,11 @@
   }
   extend(Nothing, Maybe);
```

Comment:
maybe this is just me, but shouldn't `Nothing` be a singleton? maybe it doesn't matter, as long as they compare